plugins/outputs/nsca/nsca.go

```diff
@@ -0,0 +1,167 @@
+package nsca
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+	"fmt"
+	"io"
+	"net"
+	"strconv"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/outputs"
+	"github.com/influxdata/telegraf/plugins/serializers"
+	"github.com/raff/tls-ext"
+	"github.com/raff/tls-psk"
+)
+
+// ServerInfo contains the configuration information for an NSCA server
+type ServerInfo struct {
+	// Host is the IP address or host name of the NSCA server. Leave empty for localhost.
+	Host string
+	// Port is the IP port number (no default)
+	Port string
+	// EncryptionMethod specifies the message encryption to use on NSCA messages. It defaults to ENCRYPT_NONE.
+	EncryptionMethod int
+	// Password is used in encryption.
+	Password string
+	// Timeout is the connect/read/write network timeout
+	Timeout time.Duration
+}
+
+// Message is the contents of an NSCA message
+type message struct {
+	// State is one of {STATE_OK, STATE_WARNING, STATE_CRITICAL, STATE_UNKNOWN}
+	state string
+	// Host is the host name to set for the NSCA-ng message
+	host string
+	// Service is the service name to set for the NSCA-ng message
+	service string
+	// Message is the "plugin output" of the NSCA-ng message
+	message string
+}
+
+// NSCAServer can be used as a lower-level alternative to RunEndpoint. It is NOT safe
+// to use an instance across mutiple threads.
+type NSCAServer struct {
+	Url          string
+	conn         net.Conn
+	serializer   serializers.Serializer
+	serverInfo   ServerInfo
+	CPUWarning   float64
+	CPUCritical  float64
+	DiskWarning  float64
+	DiskCritical float64
+}
+
+var sampleConfig = `
+  Url = "icinga.dev.thermeon.com:5668"
+  ## Data format to output.
+  ## Each data format has its own unique set of configuration options, read
+  ## more about them here:
+  ## https://github.com/influxdata/telegraf/blob/master/docs/DATA_FORMATS_OUTPUT.md
+  ##data_format = "influx"
+  CPUWarning = 3.6
+  CPUCritical = 4.0
+  DiskWarning = 80.0
+  DiskCritical = 90.0
+`
+
+func (n *NSCAServer) SetSerializer(serializer serializers.Serializer) {
+	n.serializer = serializer
+}
+
+func (n *NSCAServer) SampleConfig() string {
+	return sampleConfig
+}
+
+func (n *NSCAServer) Description() string {
+	return "Send telegraf measurements to nsca"
+}
+
+// define GetKey and GetIdentity methods
+func getIdentity() string {
+	return "system-checker"
```

Comment:
Isn't the identity meant to relate to the server?

Reply:
Yeah, its relate to server.now  configured.